pandas/core/exchange/column.py

```diff
@@ -0,0 +1,368 @@
+from functools import cached_property
+from typing import (
+    Any,
+    Tuple,
+)
+
+import numpy as np
+
+import pandas as pd
+from pandas.api.types import (
+    is_categorical_dtype,
+    is_string_dtype,
+)
+from pandas.core.exchange.buffer import PandasBuffer
+from pandas.core.exchange.dataframe_protocol import (
+    Column,
+    ColumnBuffers,
+    ColumnNullType,
+    DtypeKind,
+)
+from pandas.core.exchange.utils import (
+    ArrowCTypes,
+    Endianness,
+    NoBufferPresent,
+    dtype_to_arrow_c_fmt,
+)
+
+_NP_KINDS = {
+    "i": DtypeKind.INT,
+    "u": DtypeKind.UINT,
+    "f": DtypeKind.FLOAT,
+    "b": DtypeKind.BOOL,
+    "U": DtypeKind.STRING,
+    "M": DtypeKind.DATETIME,
+    "m": DtypeKind.DATETIME,
+}
+
+_NULL_DESCRIPTION = {
+    DtypeKind.FLOAT: (ColumnNullType.USE_NAN, None),
+    DtypeKind.DATETIME: (ColumnNullType.USE_NAN, None),
+    DtypeKind.INT: (ColumnNullType.NON_NULLABLE, None),
+    DtypeKind.UINT: (ColumnNullType.NON_NULLABLE, None),
+    DtypeKind.BOOL: (ColumnNullType.NON_NULLABLE, None),
+    # Null values for categoricals are stored as `-1` sentinel values
+    # in the category date (e.g., `col.values.codes` is int8 np.ndarray)
+    DtypeKind.CATEGORICAL: (ColumnNullType.USE_SENTINEL, -1),
+    # follow Arrow in using 1 as valid value and 0 for missing/null value
+    DtypeKind.STRING: (ColumnNullType.USE_BYTEMASK, 0),
+}
+
+_NO_VALIDITY_BUFFER = {
```

Comment:
Should we have validity buffer for these cases?

Reply:
No we shouldn't, as it does not make sense for the types which signal `N/A` via non-mask ways. But the spec requires us to have a `"validity"` field.

Maybe I'm misunderstanding the question?..

Reply:
I don't understand why we shouldn't have validity buffer for the dtypes. For example, let's look at a case of USE_NAN.
```python
import pandas
import numpy as np
df = pandas.DataFrame([1.0, np.nan, 3.0])
df.isna()
       0
0  False
1   True
2  False
```
I would imagine that validity buffer for the protocol column looks like this `[False, True, False]` or `[0, 1, 0]`.

Reply:
In this particular case the null property is derived from the fact that a cell contains `NaN` itself (i.e. this is conveyed via `data` buffer), so there is no point in a separate validity buffer - it would lead to extra effort and extra memory footprint for no gain.

Validity buffers make sense for `USE_BITMASK` and `USE_BYTEMASK` cases only IMHO.

Reply:
This question is out of scope for _exchange_ protocol 🙃

Reply:
Ok, will be waiting for a solution/proposal or something on this.